tremor-script/src/ast/raw.rs

```diff
@@ -1286,52 +1260,64 @@ pub enum PatchOperationRaw<'script> {
     /// we're forced to make this pub because of lalrpop
     Insert {
         /// we're forced to make this pub because of lalrpop
-        ident: ImutExprRaw<'script>,
+        ident: StringLitRaw<'script>,
```

Comment:
Why do we need to restrict this to literals?
I could imagine to use patch to manipulate a key dynamically by some application logic. For such use cases we'd need an `ImutExprInt` here.

Reply:
we can still do that using `"#{dynamic_key}"` restricting it to string literals ensurews that we don't have runtime errors for keys not being strings - we can catch this at compiletime. It also prevents confusion when someone writes:

```
patch event of
  insert timestamp => 0
end;
```

which can be easiely be read (and written) as `timestamp` being they key, not `timestamp` being the variable that holds the key.